cli/pkg/autodetect/kind.go

```diff
@@ -48,18 +42,23 @@ type ClusterDetails struct {
 	K8SVersion *version.Version
 }
 
-var (
-	ErrCannotDetectK8sVersion  = fmt.Errorf("cannot detect k8s version")
-	ErrCannotDetectClusterKind = fmt.Errorf("cannot detect cluster kind")
-)
+func GetK8SVersion() *version.Version {
+	return currentClusterDetails.K8SVersion
+}
 
-func DetectK8SClusterDetails(ctx context.Context, kc string, client *kube.Client) (ClusterDetails, error) {
+func GetClusterKind() Kind {
+	return currentClusterDetails.Kind
+}
+
+func SetK8SClusterDetails(ctx context.Context, kc string, client *kube.Client)  {
 	details := k8sutils.GetCurrentClusterDetails(kc)
 	serverVersion, err := client.Discovery().ServerVersion()
 	if err != nil {
-		return ClusterDetails{}, errors.Join(ErrCannotDetectK8sVersion, err)
+		currentClusterDetails.K8SVersion = nil
 	}
 	ver := version.MustParse(serverVersion.String())
+	currentClusterDetails.K8SVersion = ver
```

Comment:
Above, we now set `currentClusterDetails.K8SVersion = nil` and override this value here. I guess it works because `serverVersion` is empty and `MustParse` it returns nil?

Anyway, I think it's better to have this in an `else`.

Also - why `MustParse` and not `Parse` that can also report error?